tests/unifiedPlatform/pds2/pds_dataservice_test.go

```diff
@@ -355,3 +359,99 @@ var _ = Describe("{ValidatePdsHealthIncaseofFailures}", func() {
 		defer EndPDSTorpedoTest()
 	})
 })
+
+var _ = Describe("{ScaleDownScaleupPXCluster}", func() {
+	var (
+		deployment *automationModels.PDSDeploymentResponse
+		err        error
+		nodeList   []*corev1.Node
+	)
+
+	JustBeforeEach(func() {
+		StartPDSTorpedoTest("ScaleDownScaleupPXCluster", "Scales Down the PX cluster, Verify Data Services and Scales up the Px cluster", nil, 0)
+	})
+
+	It("Scales Down the PX cluster, Verify Data Services and Scales up the Px cluster", func() {
+
+		// Getting all worker nodes
+		workerNodes := node.GetWorkerNodes()
+
+		for _, ds := range NewPdsParams.DataServiceToTest {
+			Step("Deploy DataService", func() {
+				deployment, err = WorkflowDataService.DeployDataService(ds, ds.Image, ds.Version, PDS_DEFAULT_NAMESPACE)
+				log.FailOnError(err, "Error while deploying ds")
+				log.Debugf("Source Deployment Id: [%s]", *deployment.Create.Meta.Uid)
+				nodes, err := pdslib.GetNodesOfSS(*deployment.Create.Status.CustomResourceName, PDS_DEFAULT_NAMESPACE)
+				log.FailOnError(err, "Error while getting Data Serice Nodes")
+				nodeList = append(nodeList, nodes[0])
+
+			})
+
+			Step("Scale down PX Nodes on which Data Services are running", func() {
+				// disable PX Pod on the first node of each deployed Data Service
+				for _, nodeToStop := range nodeList {
+					log.InfoD("Disabling PX on Node %v ", nodeToStop.Name)
+					for _, workerNode := range workerNodes {
+						if workerNode.Name == nodeToStop.Name {
+							err = Inst().V.StopDriver([]node.Node{workerNode}, false, nil)
+							log.FailOnError(err, "error while Stopping PX on node")
+							break
+						}
+					}
+				}
+				log.InfoD("Successfully Scaled Down PX Nodes...")
+			})
+
+			log.InfoD("Sleeping 60 seconds after scale down of Px Nodes before we check the health of Data Services")
+			time.Sleep(60 * time.Second)
+
+			Step("Verify the Data Services status after scaling down the Px Nodes", func() {
+				log.InfoD("Verify the Data Services status after scaling down the Px Nodes")
+				err = WorkflowDataService.ValidatePdsDataServiceDeployments(*deployment.Create.Meta.Uid, ds, ds.Replicas, WorkflowDataService.PDSTemplates.ResourceTemplateId, WorkflowDataService.PDSTemplates.StorageTemplateId, PDS_DEFAULT_NAMESPACE, ds.Version, ds.Image)
+				log.FailOnError(err, "Error while Validating dataservice after px nodes scale down")
+			})
+
+			stepLog := "Running Workloads after scale down of PX Nodes"
+			Step(stepLog, func() {
+				_, err := WorkflowDataService.RunDataServiceWorkloads(*deployment.Create.Meta.Uid)
+				log.FailOnError(err, "Error while running workloads on ds")
+			})
+
+			Step("Scale Up PX  Nodes", func() {
+				for _, nodeToStart := range nodeList {
```

Comment:
same like above, a method to start the px on the node.

Reply:
Done